### ISSUE TITLE
Rancher cli v0.1.0 is unable to exec into container

### ISSUE THREAD

**Issue by sra** [rancher/rancher#5797]
**Rancher Version:**
1.1.2

**Docker Version:**
1.11

**OS and where are the hosts located? (cloud, bare metal, etc):**
Both boot2docker and AWS, different installations same exact message

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher servers

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**

```
$ rancher-0.1.0 exec -it appname bash
An error occurred trying to connect: Post http://%2Fvar%2Ffolders%2Fk_%2F7k3hg40j2jldqbq24w_b8b7h0000gn%2FT%2Fdocker-sock788202943/v1.22/containers/df154460b67f1b1eb7e64f1b6b54de1ddd921745a925ad4514517972ac09f75a/exec: EOF
ERRO[0000] Failed to handle connection: Post http://192.168.99.100:8090/v1/projects/1a5/hosts/1h2/?action=dockersocket: net/http: invalid header field value "\x04" for key Content-Length

$ rancher-0.0.2 exec -it appname bash
bash$
```

**Results:** 

**Expected:**

Expect to be able to exec into container as with v0.0.2

**Comment by whiteadam** [comment]
I think I also have this issue.

**Comment by PaulLiang1** [comment]
Had this issue as well 

```
ERRO[0001] Failed to handle connection: Post http://rancher-server01/v1/projects/1111/hosts/2222/?action=dockersocket: net/http: invalid header field value "\x04" for key Content-Length
```

**Comment by deniseschannon** [comment]
Unable to reproduce with v0.3.1 in a non-auth setup and a github enabled setup. 

Please re-open if you have issues with a v0.3.1 and a post v1.2.0-pre4-rc11 release. 

We would prefer if you waited for v1.2.0 for Rancher. 

Note: v0.3.1 doesn't work with 1.1.4.

**Comment by joshwget** [comment]
Able to reproduce with Go 1.7 on Linux - not sure why this is occurring on Go 1.6 based builds, but there's certainly some issue here.

**Comment by joshwget** [comment]
I was wrong, accidentally used an earlier commit. This should be fixed in v0.3.1. Sorry @deniseschannon :smile:
